internal/kms/keyprotect.go

```diff
@@ -62,6 +63,21 @@ var _ = RegisterProvider(Provider{
 	Initializer: initKeyProtectKMS,
 })
 
+// RegisterProvider for kmsTypeKeyProtectMetadataOld is kept here for backward compatibility.
+var _ = RegisterProvider(Provider{
+	UniqueID:    kmsTypeKeyProtectMetadataOld,
+	Initializer: initKeyProtectKMSOld,
+})
+
+// initKeyProtectKMSOld is the wrapper with a warning log.
+func initKeyProtectKMSOld(args ProviderInitArgs) (EncryptionKMS, error) {
+	log.WarningLog(context.TODO(), "%s is deprecated provider for IBM key Protect,"+
```

Comment:
use `WarningLogMsg` as no actual context is passed here.

Reply:
@Madhu-1 done. thanks. ptal.